examples/batch-tx.js

```diff
@@ -0,0 +1,379 @@
+        .setLogger(new Logger(LogLevel.Info));
+    const aliceBatchedTransfer = await new TransferTransaction()
+        .addHbarTransfer(alice, Hbar.from(-1, HbarUnit.Hbar))
+        .addHbarTransfer(
+            client.getOperator().accountId,
+            Hbar.from(1, HbarUnit.Hbar),
+        )
+        .batchify(aliceClient, batchKey1);
+
+    console.log("Created first account (Alice): " + alice.toString());
+
+    const bobKey = PrivateKey.generateECDSA();
+    const bob = (
+        await (
+            await new AccountCreateTransaction()
+                .setKeyWithoutAlias(bobKey)
+                .setInitialBalance(new Hbar(2))
+                .execute(client)
+        ).getReceipt(client)
+    ).accountId;
+
+    const bobClient = Client.forName(process.env.HEDERA_NETWORK).setOperator(
+        bob,
+        bobKey,
+    );
+
+    const bobBatchedTransfer = await new TransferTransaction()
+        .addHbarTransfer(bob, Hbar.from(-1, HbarUnit.Hbar))
+        .addHbarTransfer(
+            client.getOperator().accountId,
+            Hbar.from(1, HbarUnit.Hbar),
+        )
+        .batchify(bobClient, batchKey2);
+
+    console.log("Created second account (Bob): " + bob.toString());
+
+    const carolKey = PrivateKey.generateECDSA();
+    const carol = (
+        await (
+            await new AccountCreateTransaction()
+                .setKeyWithoutAlias(carolKey)
+                .setInitialBalance(new Hbar(2))
+                .execute(client)
+        ).getReceipt(client)
+    ).accountId;
+
+    const carolClient = Client.forName(process.env.HEDERA_NETWORK).setOperator(
+        carol,
+        carolKey,
+    );
+    const carolBatchedTransfer = await new TransferTransaction()
+        .addHbarTransfer(carol, Hbar.from(-1, HbarUnit.Hbar))
+        .addHbarTransfer(
+            client.getOperator().accountId,
+            Hbar.from(1, HbarUnit.Hbar),
+        )
+        .batchify(carolClient, batchKey3);
+
+    console.log("Created third account (Carol): " + carol.toString());
+
+    /**
+     * Step 3:
+     * Get the balanace in order to compete after this batch
+     */
+    const aliceBalanceBefore = await new AccountBalanceQuery()
+        .setAccountId(alice)
+        .execute(client);
+    var bobBalanceBefore = await new AccountBalanceQuery()
+        .setAccountId(bob)
+        .execute(client);
+    var carolBalanceBefore = await new AccountBalanceQuery()
+        .setAccountId(carol)
+        .execute(client);
+    var operatorBalanceBefore = await new AccountBalanceQuery()
+        .setAccountId(client.getOperator().accountId)
+        .execute(client);
+
+    /**
+     * Step 4:
+     * Execute the batch transaction
+     */
+
+    console.log("Executing batch transaction...");
+    const batch = await (
+        await (
+            await new BatchTransaction()
+                .addInnerTransaction(aliceBatchedTransfer)
+                .addInnerTransaction(bobBatchedTransfer)
```

Comment:
Don't we need to set `setBatchKey(Key)` here?

Reply:
and its not on the batch transaction it is on the inner transaction so nvm thanks!